runtime/v2/shim/shim.go

```diff
@@ -133,6 +141,7 @@ func run(id string, initFunc Init) error {
 		return fmt.Errorf("shim namespace cannot be empty")
 	}
 	ctx := namespaces.WithNamespace(context.Background(), namespaceFlag)
+	ctx = context.WithValue(ctx, OptsKey{}, Opts{Debug: debugFlag})
```

Comment:
It only looks like this Opts is used in the tests, is this needed still?

Reply:
https://github.com/containerd/containerd/pull/2686/files#diff-b6be0a426cab99fb4e2d6ff6bf99fbf9R236

It is used in the `StartShim` command. I was actually surprised this wasn't already plumbed but it seems that if a `--debug` is sent to a `<shim>.exe --debug start` command the re-invoke will drop the `--debug` because it has no debug context. So this forwards the debug context to the `StartShim` so that the shim implementation can adhere to it. I did not fix it for the `runc` shim though.

Reply:
Ok.  This feels a little out of place right now, but lets continue and develop out the tracing feature and see where this ends up.

Reply:
Sounds good to me its easy to take out later